### PR TITLE
Break apart the setup scripts and support Big Sur

### PR DESCRIPTION
**Pull request by willbarton**
This PR does three major things:

- Breaks up out setup scripts so they can be run individually:
   - `homebrew_setup.sh` to install and configure [Homebrew](http://brew.sh/) 
   - `python_setup.sh` to install and configure [our Python development environment](# https://github.com/cfpb/development/blob/master/guides/installing-python.md)
   - `node_setup.sh` to install and configure our Node development environment
   - `git_setup.sh` to install and configure git and [git-secrets](https://github.com/cfpb/development/blob/master/tools/git-secrets-patterns/README.md)
   - `dotfiles_setup.sh` to apply our standard dotfiles
- Supports building Python on Big Sur
- Switches the environment setup from bash to zsh

### CODE REVIEW

**Comment by flacoman91** [comment]
I wanted to run this and reset my env, but it was failing in the homebrew setup because I had a prior installation.  After I manually deleted the homebrew directory in my home folder,  it seems to continue.

File: dotfiles_setup.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+printf "\n\n$(tput bold)$(tput setaf 7)$(tput setab 2)Creating CFPB standard dotfiles$(tput sgr 0)"
```
Comment (flacoman91) [inline dotfiles_setup.sh:3]: can you put a newline between  the message here.  I see this:
Screen Shot 2021-06-29 at 12 36 10 PM

File: python_setup.sh [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+install_pyenv() {
+    echo "$(tput setaf 3)$(tput setab 4)...Installing pyenv$(tput sgr 0)"
+
+    brew install pyenv pyenv-virtualenv
+
+    echo "# Python and pyenv" >> ${HOME}/.zshenv
+    echo export PYENV_ROOT="${HOME}/.pyenv"
+    echo export PATH="$PYENV_ROOT/bin:$PATH"
+    echo eval "$(pyenv init --path)" >> ${HOME}/.zshenv
+    echo eval "$(pyenv virtualenv-init -)" >> ${HOME}/.zshenv
```
Comment (higs4281) [inline python_setup.sh:18]: A few fixes to these setups got a Big Sur user going:

```suggestion
    echo "# Python and pyenv" >> ${HOME}/.zshenv
    echo export PYENV_ROOT="${HOME}/.pyenv" >> ${HOME}/.zshenv
    echo export PATH="$PYENV_ROOT/bin:$PATH" >> ${HOME}/.zshenv
    echo eval "$(pyenv init --path)" >> ${HOME}/.zshenv
    echo eval "$(pyenv init -)" >> ${HOME}/.zshenv
    echo eval "$(pyenv-virtualenv init -)" >> ${HOME}/.zshenv
```

File: python_setup.sh [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    echo export PYENV_ROOT="$HOME/.pyenv"
+eval "$(pyenv init --path)"
```
Comment (higs4281) [inline python_setup.sh:21]:
```suggestion

```

File: dotfiles_setup.sh (hunk 1 above)
Comment (willbarton) [reply dotfiles_setup.sh]: Done in 82c2ad1

**Comment by willbarton** [comment]
@higs4281 @flacoman91 thanks! I've committed all the suggestions. Should we move this out of draft?

**Comment by Scotchester** [comment]
What, you found my echoes a bit too yell-y? 😁

**Comment by higs4281** [comment]
> What, you found my echoes a bit too yell-y? 😁

Please drive quietly in the Big Sur.

**Review by higs4281 — COMMENTED** [review]
The preflight script succeeds for me, and would be a good backstop. I couldn't test the failure state effectively while on subnet. I tried turning off wifi, but the script just fails to exit.

**Comment by willbarton** [comment]
I changed the preflight script to require the manual step of `export DEV_HOSTS=<hostnames>`. If this all looks good, I'll update the documentation a bit more too.

Are there any other preflight checks we should do?  @higs4281 or anyone else? Xcode and existing on the dev subnet are the things we have in our onboarding docs.

**Comment by higs4281** [comment]
A couple thoughts about making pyenv friendlier.

Some developers may be used to using virtualenvwrapper's `workon` command, and this alias would come in handy for them:

```bash
alias workon='pyenv activate $1 > /dev/null 2> /dev/null'
```

Also, I've been wondering if we could make life easier by creating a cfgov env with our current version at the end of python_setup.sh?

```bash
pyenv virtualenv ${PYTHON3_VER} cfgov
```

**Comment by willbarton** [comment]
I like the idea of the `workon` alias (I admit, my fingers still type `workon` before I catch them).

I do wonder... I'd really love to make the cfgov setup easier generally. I wonder if we should hold off on bringing it further into these scripts for now?

**Comment by higs4281** [comment]
> hold off ...

That makes sense to me. I was just thinking about extra steps recent onboarders had to take, but it would be good to polish our approach.

**Review by higs4281 — APPROVED** [review]
Onward!

**Pull request merged**